src/main/java/com/google/devtools/build/lib/remote/disk/DiskCacheClient.java

```diff
@@ -329,12 +332,14 @@ private void saveFile(Digest digest, Store store, InputStream in) throws IOExcep
     Path temp = getTempPath();
 
     try {
-      try (FileOutputStream out = new FileOutputStream(temp.getPathFile())) {
-        ByteStreams.copy(in, out);
-        // Fsync temp before we rename it to avoid data loss in the case of machine
-        // crashes (the OS may reorder the writes and the rename).
-        out.getFD().sync();
-      }
+      File outFile = temp.getPathFile();
+      Files.copy(in, outFile.toPath(), StandardCopyOption.REPLACE_EXISTING);
+      // FIXME: Do we need it?
+      // try (FileOutputStream out = new FileOutputStream(outFile)) {
+      //   // Fsync temp before we rename it to avoid data loss in the case of machine
+      //   // crashes (the OS may reorder the writes and the rename).
+      //   out.getFD().sync();
+      // }
```

Comment:
We need it unless you can convince me that the scenario described in the comment is not a concern (https://danluu.com/deconstruct-files/ is a good reference on what can go wrong when a system crashes in the middle of a filesystem operation).

Reply:
I looked at the NIO source code. Only `public static Path copy(Path source, Path target, CopyOption... options)
throws IOException` can optimize the IO operation, and there is no fsync method available.